tests/units/adapters/test_data_saver.py

```diff
@@ -40,15 +41,16 @@ def html_writer(tmpdir):
 @pytest.fixture(params=['txt', 'html'])
 def template_name_fixture(request) -> str:
     if request.param == 'txt':
-        return str(TEMPLATE_PATH / 'student_report_template.txt')
+        return str('student_report_template.txt')
     elif request.param == 'html':
-        return str(TEMPLATE_PATH / 'student_report_template.html')
+        return str('student_report_template.html')
 
 
-@pytest.mark.parametrize('expected_content', [
-    (TEMPLATE_PATH / 'student_report_template.txt').read_text(),
-    (TEMPLATE_PATH / 'student_report_template.html').read_text()
-])
+@pytest.mark.parametrize('expected_content',
+                         [
+                             (TEMPLATE_PATH_DATA / 'test_data_student.txt').read_text(),
+                             (TEMPLATE_PATH_DATA / 'test_data_student.html').read_text()
```

Comment:
As mentioned before, test scenario parameters defined in `pytest.mark.parametrize` should not perform any I/O operations. And `read_text()` does it. So you should leave here file path only (and rename variable from `expected_content` to `expected_content_path` and within test function add `expected_content = expected_content_path.read_text()`